_source/_posts/2023-07-28-scim-workshop.md

```diff
@@ -0,0 +1,1144 @@
+
+#### Deprovision a user
+
+Let's say Tom decides to leave Portal, so we need to deprovision him from the application. In the Assignments tab of the Okta SCIM application, use the blue X next to Tom's entry to unassign him from the app. This unassignment makes Okta send the Todo App PATCH request, setting the unassigned user's `active` attribute to `false`. This indicates that a user's account has been suspended. We can confirm that Tom's `active` attribute is now `false` in the Todo app's database through the Prisma web interface.
+
+#### Reprovision a user
+
+Let's say Tom later decides to return to Portal and needs access to this Todo app again. To reactivate Tom's account, we will repeat the steps for assigning his Okta account to the application.  Re-activating Tom causes Okta to send a PATCH request to our app, setting his  `active` attribute to `true`. Again, we can confirm that Tom's account is now `active`  through our Prisma database web interface.
+
+#### Change a user's info in Okta
+
+Let's look at one more scenario. Let's say not only has Tom returned but also changed his name! Worry not, as our SCIM connection can handle this profile update. Let's change Tom's name to Leo and watch SCIM propagate the new name to our Todo app. 
+
+In the Directory tab of the Okta admin console sidebar, navigate to People, and click on Tom's name in the user list to edit his settings. Under Profile, click "Edit" in the Attributes pane. Change the `firstName` field to Leo, and update his `login` and `email` to `leo.anderson@portal.example`. Use the blue Save button at the bottom of the page to save your changes. 
+
+In your Todo app logs, what request do you expect to see from Okta when a user's information is updated? You should see the PUT request to your `/scim/v2/Users/4` endpoint shortly after saving Leo's new name. 
+
+
+## Tool Recommendations for Development 
+While troubleshooting and developing an application, it can be helpful to use tools that let you inspect your code's behavior closely, including the way it handles secrets. These tools helped me as I developed the sample code used in the workshop, and might help you as you build your own projects. 
+
+The features that make these tools excellent for testing also make them terrible for use in production. In a production context, security is essential. For troubleshooting production services, integrate your application with your team's secure logging and secrets management infrastructure. 
+
+### View requests to and from the app
+When [ngrok](https://ngrok.com/) is supported in my environment, it's my favorite tool for viewing requests to and from a local server. If ngrok is blocked, use Okta's [system logs](https://help.okta.com/en-us/Content/Topics/Reports/Reports_SysLog.htm) to see requests from the app to Okta. [Morgan](https://github.com/expressjs/morgan) can help you log inbound HTTP requests to the server.
```

Comment:
It seems weird that previously localtunnel was used, and now this says ngrok is preferred?

Reply:
@aaronpk, through recording the video, I experienced multiple and sporadic timeout issues - https://github.com/localtunnel/localtunnel/issues/612. And sadly, I was not able to use ngrok through our corporate network 😞 .

Reply:
```suggestion
When [ngrok](https://ngrok.com/) is supported in my environment, it's my favorite tool for viewing requests to and from a local server. Unfortunately, some firewalls forbid ngrok but allow localtunnel, which is why localtunnel was used for this workshop's demonstrations. If ngrok is blocked, use Okta's [system logs](https://help.okta.com/en-us/Content/Topics/Reports/Reports_SysLog.htm) to see requests from the app to Okta. [Morgan](https://github.com/expressjs/morgan) can help you log inbound HTTP requests to the server.
```